lib/beaker/options/subcommand_options_file_parser.rb

```diff
@@ -3,18 +3,30 @@ module Options
     #A set of functions to read options files
     module SubcommandOptionsParser
 
+      HOMEDIR_OPTIONS_FILE_PATH = ENV['HOME']+'/.beaker/subcommand_options.yaml'
+
+      def self.parse_options_file(options_file_path)
+        result = OptionsHash.new
+        if File.exist?(options_file_path)
+          result = YAML.load_file(options_file_path)
+        end
+        result
+      end
+
       # @return [OptionsHash, Hash] returns an empty OptionHash or loads subcommand options yaml
       #   from disk
-      def self.parse_subcommand_options(argv)
+      def self.parse_subcommand_options(argv, home_dir=false)
```

Comment:
So instead of passing in a boolean here to determine whether or not we should read from the home directory, I would pass in the file to read from as a string, e.g. `parse_subcommand_options(argv, options_file)`. That way, we don't need the `if` branch with the early return statement, but rather, we can just do `result = SubcommandOptionsParser.parse_options_file(options_file)`, and then, we can change line 221 of parser.rb to be something like `homedir_options = Beaker::Options::SubcommandOptionsParser.parse_subcommand_options(args, ENV['HOME']+'/.beaker/subcommand_options.yaml')`, and change line 226 of parser.rb to something like: `subcommand_options = Beaker::Options::SubcommandOptionsParser.parse_subcommand_options(args, Beaker::Subcommands::SubcommandUtil::SUBCOMMAND_OPTIONS)`.